cpp/betweenness_centrality_module/betweenness_centrality_online_module.cpp

```diff
@@ -0,0 +1,249 @@
+bool initialized = false;
+}  // namespace
+
+void InsertOnlineBCRecord(mgp_graph *graph, mgp_result *result, mgp_memory *memory, const std::uint64_t node_id,
+                          double bc_score) {
+  auto *record = mgp::result_new_record(result);
+
+  mg_utility::InsertNodeValueResult(graph, record, kFieldNode.data(), node_id, memory);
+  mg_utility::InsertDoubleValueResult(record, kFieldBC.data(), bc_score, memory);
+}
+
+void InsertMessageRecord(mgp_result *result, mgp_memory *memory, const char *message) {
+  auto *record = mgp::result_new_record(result);
+
+  mg_utility::InsertStringValueResult(record, kFieldMessage.data(), message, memory);
+}
+
+void Set(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    const auto normalize = mgp::value_get_bool(mgp::list_at(args, 0));
+    auto threads = mgp::value_get_int(mgp::list_at(args, 1));
+
+    if (threads <= 0) threads = std::thread::hardware_concurrency();
+
+    auto graph = mg_utility::GetGraphView(memgraph_graph, result, memory, mg_graph::GraphType::kUndirectedGraph);
+    const auto node_bc_scores = algorithm.Set(*graph, DEFAULT_DIRECTED, normalize, threads);
+    ::initialized = true;
+
+    for (const auto [node_id, bc_score] : node_bc_scores) {
+      InsertOnlineBCRecord(memgraph_graph, result, memory, node_id, bc_score);
+    }
+  } catch (const std::exception &e) {
+    mgp::result_set_error_msg(result, e.what());
+    return;
+  }
+}
+
+void Get(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    const auto normalize = mgp::value_get_bool(mgp::list_at(args, 0));
+
+    auto graph = mg_utility::GetGraphView(memgraph_graph, result, memory, mg_graph::GraphType::kUndirectedGraph);
+    const auto node_bc_scores = algorithm.Get(*graph, normalize);
+
+    for (const auto [node_id, bc_score] : node_bc_scores) {
+      InsertOnlineBCRecord(memgraph_graph, result, memory, node_id, bc_score);
+    }
+  } catch (const std::exception &e) {
+    mgp::result_set_error_msg(result, e.what());
+    return;
+  }
+}
+
+void Update(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    const auto created_nodes = mgp::value_get_list(mgp::list_at(args, 0));
+    const auto created_edges_list = mgp::value_get_list(mgp::list_at(args, 1));
+    const auto deleted_nodes = mgp::value_get_list(mgp::list_at(args, 2));
+    const auto deleted_edges_list = mgp::value_get_list(mgp::list_at(args, 3));
+    const auto normalize = mgp::value_get_bool(mgp::list_at(args, 4));
+    const auto threads = mgp::value_get_int(mgp::list_at(args, 5));
+
+    auto graph = mg_utility::GetGraphView(memgraph_graph, result, memory, mg_graph::GraphType::kUndirectedGraph);
+    std::unordered_map<uint64_t, double> node_bc_scores;
+
+    if (!::initialized) {
+      node_bc_scores = algorithm.Set(*graph, DEFAULT_DIRECTED, normalize, threads);
+      ::initialized = true;
```

Comment:
Initialized flag should be implementation detail, move it into algorithm object

Reply:
Agreed, will move so that each part has clearly defined responsibilities.